xbmc/guilib/guiinfo/AddonsGUIInfo.cpp

```diff
@@ -213,7 +213,9 @@ bool CAddonsGUIInfo::GetBool(bool& value, const CGUIListItem *gitem, int context
       value = true;
       const CFileItem* item = static_cast<const CFileItem*>(gitem);
       if (item->GetAddonInfo())
-        value = CServiceBroker::GetAddonMgr().IsAutoUpdateable(item->GetAddonInfo()->ID());
+        value = !(!CServiceBroker::GetAddonMgr().IsAutoUpdateable(item->GetAddonInfo()->ID()) &&
+                  CServiceBroker::GetAddonMgr().IsAddonInstalled(item->GetAddonInfo()->ID(),
+                                                                 item->GetAddonInfo()->Origin()));
```

Comment:
The double negation is confusing. Why not an `||` instead?

Reply:
didn‘t want to use `||` because then both calls are executed for any list entry. this way we call `IsAddonInstalled` only for the pinned.

Reply:
but i guess `IsAutoUpdateable || !IsAddonInstalled` is fine in this regard too..

Reply:
Which do you find easier to read?

Reply:
the `||` of course. was just completely convinced that a short-circuit does not exist with || for some reason 🤫

Reply:
This needs a comment explaining the logic as it’s hard to figure it out without it. So explain why the not installed origin check is important.